### PR TITLE
Remove CIDR class using apache common net instead

### PR DESCRIPTION
**Pull request by Astralidea**
The current CIDR class implementation has some bugs.
The implementation is actually similar to Apache Common Net Library.
So we considered dropping the CIDR class and going straight to Apache Common Net Library.

In the case of writing IP directly, we simply do string matching,
 and in the case of CIDR format, we use the library for interval judgment

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/service/FrontendOptions.java [hunk 1]
```diff
@@ -129,15 +129,22 @@ private static void analyzePriorityCidrs() {
...
-                return true;
+        for (String cidr : priorityCidrs) {
+            cidr = cidr.trim();
+            if (!cidr.contains("/")) {
+                // it is not valid CIDR, compare ip directly.
+                if (cidr.equals(ip.trim())) {
+                    return true;
+                }
+            } else {
+                SubnetUtils.SubnetInfo subnetInfo = new SubnetUtils(cidr).getInfo();
```
[the remark this answers is not in the document]
Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/service/FrontendOptions.java:144]: YES. SubnetUtils will check CIDR format。

[the remark this answers is not in the document]
Comment (imay) [reply fe/fe-core/src/main/java/com/starrocks/service/FrontendOptions.java:144]: When an invalid format is passed as input, what's the SubnetUtils action. If it throws an exception, the exception should be catch. If it returns null, null should be checked.

[the remark this answers is not in the document]
Comment (imay) [reply fe/fe-core/src/main/java/com/starrocks/service/FrontendOptions.java:144]: Thanks, I got it.

**Review by imay — DISMISSED** [review]

**Review by wyb — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/service/FrontendOptions.java [hunk 2]
```diff
@@ -129,15 +129,22 @@ private static void analyzePriorityCidrs() {
...
 
     private static boolean isInPriorNetwork(String ip) {
-        for (CIDR cidr : priorityCidrs) {
-            if (cidr.contains(ip)) {
-                return true;
+        for (String cidr : priorityCidrs) {
+            cidr = cidr.trim();
+            if (!cidr.contains("/")) {
+                // it is not valid CIDR, compare ip directly.
+                if (cidr.equals(ip.trim())) {
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/service/FrontendOptions.java:140]: should be case insensitive or not ?

Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/service/FrontendOptions.java:140]: this input only have number and not. it does not need to consider case insensitive.

**Review by imay — APPROVED** [review]

**Comment by wyb** [comment]
Please check if BE has similar bugs

**Review by chaoyli — APPROVED** [review]

**Pull request merged**